### ISSUE TITLE
Allow CompletionOptions to provide additional data

### ISSUE THREAD

**Issue by tinaschrepfer** [Microsoft/language-server-protocol#734]
Currently server can specify trigger characters through CompletionOptions.  The following additional data would also be helpful in creating a more desired IntelliSense experience:

* commit characters - i.e. aside from Enter, what characters when typed should commit the completion item selected.  This could be the same as trigger character, a subset, or something completely different.
* dismiss characters - i.e. aside from Escape, what characters when typed should dismiss the completion session?  An example of this would be semicolon.

Another idea @LukaszMendakiewicz had was that some of this data can be overridden for a given completion session.  For example, a server can define default values for this during initialization, but each completion request could return different sets of commit and dismiss characters.

**Comment by tinaschrepfer** [comment]
It looks like the completion item returned has commit characters, but we would still need a way to know what dismiss characters are.

**Comment by dbaeumer** [comment]
Correct, both the options and the completion item support commit characters.
